### PR TITLE
fixing dist tc compilation error due to moderngpu

### LINKED ISSUE #148: lonestardist/tc doesn't build

### PR DESCRIPTION
**Pull request by vishweshjatala**
Replaced the older moderngpu function calls with new ones. I have tested locally and verified the correctness.  

Closes #148

### CODE REVIEW

**Comment by ddn0** [comment]
Closes #148 

Can you also re-enable tc for the all target (in `lonestardist/CMakeLists.txt`)?

File: lonestardist/tc/tc_cuda.cu [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
-#include "kernels/reduce.cuh"
+#include "moderngpu/kernel_reduce.hxx"
 #include "tc_cuda.cuh"
-#include "kernels/segmentedsort.cuh"
-#include "moderngpu.cuh"
-#include "util/mgpucontext.h"
+#include "moderngpu/kernel_segsort.hxx"
 #include <cuda_profiler_api.h>
-mgpu::ContextPtr mgc;
+using namespace mgpu;
```
Comment (ddn0) [inline lonestardist/tc/tc_cuda.cu:13]: I generally like explicit namespaces (i.e., avoiding `using namespace XYZ`) as it is easier to see where symbols come from.

In this case, we still explicitly refer to `cub` like in `cub::BlockReduce` below. Is there a reason to use the `mpgu` namespace but not `cub`?

**Comment by insertinterestingnamehere** [comment]
@vishweshjatala can we get this merged soon?

File: lonestardist/tc/tc_cuda.cu (hunk 1 above)
Comment (vishweshjatala) [reply lonestardist/tc/tc_cuda.cu:13]: Sure, we can use the explicit namespace. I will fix the code.

**Comment by vishweshjatala** [comment]
@insertinterestingnamehere , I fixed the issues. You can merge if you are fine.

**Comment by insertinterestingnamehere** [comment]
LGTM. Will merge once the tests pass. Thanks!

**Pull request merged**